library/pkcs12.c

```diff
@@ -186,13 +189,18 @@ static void pkcs12_fill_buffer( unsigned char *data, size_t data_len,
     unsigned char *p = data;
     size_t use_len;
 
-    while( data_len > 0 )
+    if( filler != NULL && fill_len != 0 )
     {
-        use_len = ( data_len > fill_len ) ? fill_len : data_len;
-        memcpy( p, filler, use_len );
-        p += use_len;
-        data_len -= use_len;
+        while( data_len > 0 )
+        {
+            use_len = ( data_len > fill_len ) ? fill_len : data_len;
+            memcpy( p, filler, use_len );
+            p += use_len;
+            data_len -= use_len;
+        }
     }
+    else
+        memset( data, 0, data_len );
```

Comment:
This doesn't look right: why fill `data` with null bytes? PKCS#12 just concatenates strings here, and if the input is empty, so is the output.

In fact `pkcs12_fill_buffer` should probably require `fill_len != 0`.

Reply:
My motivation was as below: If no input, the output buffer should be 'empty'. I have worked around this in other ways, so removed.